File: tests/uasm/sample/method_chain_numeric/MethodChainNumericTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class MethodChainNumericTest : UdonSharpBehaviour
{
    private int AddOne(int value)
    {
        return value + 1;
    }

    private int MultiplyByTwo(int value)
    {
        return value * 2;
    }

    private int Clamp(int value, int min, int max)
    {
        if (value < min) return min;
        if (value > max) return max;
        return value;
    }

    public void Start()
    {
        int result = Clamp(MultiplyByTwo(AddOne(5)), 0, 20);
        Debug.Log(result);
    }
}

File: tests/uasm/sample/method_chain_numeric/MethodChainNumericTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class MethodChainNumericTest extends UdonSharpBehaviour {
  private addOne(value: number): number {
    return value + 1;
  }

  private multiplyByTwo(value: number): number {
    return value * 2;
  }

  private clamp(value: number, min: number, max: number): number {
    if (value < min) return min;
    if (value > max) return max;
    return value;
  }

  Start(): void {
    const result = this.clamp(this.multiplyByTwo(this.addOne(5)), 0, 20);
    Debug.Log(result);
  }
}
